ravenframework/Steps/MultiRun.py

```diff
@@ -267,6 +267,11 @@ def _localTakeAstepRun(self, inDictionary):
     # END while loop that runs the step iterations (collection and submission-for-DET)
     # if any collected runs failed, let the sampler treat them appropriately, and any other closing-out actions
     sampler.finalizeSampler(self.failedRuns)
+    for myLambda, outIndex in self._outputCollectionLambda:
+      if isinstance(outputs[outIndex], OutStreamEntity):
+        myLambda([None,outputs[outIndex]])
+        self.raiseAMessage(f'Finalized output "{inDictionary["Output"][outIndex].name}"')
```

Comment:
@alfoa The proposed changes can resolve the issue. However, it can be confusing since these lines are almost similar to collection part in the same function. Could you provide more details why previous collection can not collect the final solution? Is it possible to make some modifications inside the optimizer to enable it? I have two concerns for the proposed approach: 
1) Two collections in the same function, which make it very confusing. Either add more explanations or find a way to avoid it.
2) It is also very confusing in the FilePrint.py, since the new added lines to check the rlzIndex seem very unnecessary.  I see the changes make the final collection possible, but it is really hard to understand why these lines are needed unless the developers fully understand the collections in the steps.

Reply:
The lines in the rlzIndex are not unnecessary. They are required when the collection in triggered on <Output> data objects that are not "collected/created" by the Optimizers. 

Basically, the "SolutionExport" in the Optimizers is "updated" with the final solution **after** the collection is triggered (at the begin of the processing of the "last job"). So the Outstream is not invoked before exiting the Multirun loop. 

This modification was the "minimal viable" solution to trigger an out stream call at the end of the multi run. 

Maybe another approach could be to split the calls to the output collection:
- Right after a Job is finished for data objects (DataObjects/Databases)
- After the call to the ```finalizeActualSampling``` for OutStreams